api/envoy/config/route/v3/route_components.proto

```diff
@@ -902,20 +902,25 @@ message RouteAction {
 
   oneof host_rewrite_specifier {
     // Indicates that during forwarding, the host header will be swapped with
-    // this value.
+    // this value. The original value of the host header will be appended to the
+    // :ref:`config_http_conn_man_headers_x-forwarded-host` header.
```

Comment:
I'm inclined to wonder if which way this should default. It seems no one has needed it so far, so making it optional and default-off seems both in line with our new API rules (cc @markdroth) and save bytes on the wire.

If we opt to leave it as is, we definitely need to runtime guard (as mentioned in CONTRIBUTING.md) but I'd hold off on guarding until we settle on config.